core/kernel/dt.c

```diff
@@ -553,23 +554,35 @@ void init_external_dt(unsigned long phys_dt)
 	if (!IS_ENABLED(CFG_EXTERNAL_DT))
 		return;
 
-	if (!phys_dt) {
+	if (IS_ENABLED(CFG_TRANSFER_LIST))
```

Comment:
There are dependencies `CFG_EXTERNAL_DT=y`. Should _mk/config.mk_ enforce that switch when `CFG_TRANSFER_LIST` is enabled?

Reply:
Add enforcing `CFG_EXTERNAL_DT` when `CFG_TRANSFER_LIST` is defined.